app/models/enterprise.rb

```diff
@@ -220,6 +230,11 @@ def relatives
     ", self.id, self.id)
   end
 
+  def relatives_and_oc_producers(order_cycles)
+    enterprise_ids = Exchange.in_order_cycle(order_cycles).incoming.pluck :sender_id
```

Comment:
I find that I have to read and understand the query before I understand what `enterprise_ids` is. Maybe we could overcome that by naming the var more descriptively, possibly something like `producer_ids` or `oc_producer_ids`?